ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs

```diff
@@ -342,25 +342,41 @@ type LocalHandle = Socket
 #endif
 
 -- | System dependent LocalSnocket type
+--
+#if defined(mingw32_HOST_OS)
+data LocalSocket = LocalSocket { getLocalHandle :: LocalHandle
+                               , getLocalPath   :: LocalAddress
+                               }
+    deriving (Eq, Generic)
+    deriving Show via Quiet LocalSocket
+#else
 newtype LocalSocket  = LocalSocket { getLocalHandle :: LocalHandle }
     deriving (Eq, Generic)
     deriving Show via Quiet LocalSocket
+#endif
 
 -- | System dependent LocalSnocket
 type    LocalSnocket = Snocket IO LocalSocket LocalAddress
 
-localSnocket :: IOManager -> FilePath -> LocalSnocket
+
+-- | Create a 'LocalSnocket'.
+--
+-- On /Windows/, there is no way to get path associated to a named pipe.  To go
+-- around this, the address passed to 'open' via 'LocalFamily' will be
+-- referenced by 'LocalSocket'.
+--
+localSnocket :: IOManager -> LocalSnocket
 #if defined(mingw32_HOST_OS)
-localSnocket ioManager path = Snocket {
-      getLocalAddr  = \_ -> return localAddress
-    , getRemoteAddr = \_ -> return localAddress
-    , addrFamily  = \_ -> LocalFamily
+localSnocket ioManager = Snocket {
+      getLocalAddr  = return . getLocalPath
+    , getRemoteAddr = return . getLocalPath
```

Comment:
Reusing the local path as the remote path doesn't make for a unique connection id. I think we need to store the counter path we use for `accept` here too. Otherwise the address return from `accept` won't match subsequent calls to getRemoteAddr.

Reply:
Then we need to store the counter inside the `LocalSnocket`, let's see if this will work.

Reply:
I think it will be unique: check [this](https://github.com/input-output-hk/ouroboros-network/pull/3354/files/f755ea5eabf0ef76b2ea3b82ce063d7fd14b3fa1?file-filters%5B%5D=.hs#diff-b89417778eab9a61bc587228ba77a40615f2a501aae28864cbf631139769f7d7R464-R465).

But it does not seem right that we use `getLocalPath` for both local and remote.  We could store both: unmodified path (to be used by `getLocalAddr) and modified one (for `getRemoteAddr`).

Reply:
>But it does not seem right that we use getLocalPath for both local and remote. We could store both: unmodified path (to be used by getLocalAddr) and modified one (for getRemoteAddr`).
Yes.